Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm

```diff
@@ -29,9 +29,70 @@
 #import "CoreIPCSecureCoding.h"
 
 #import "ArgumentCodersCocoa.h"
+#import "AuxiliaryProcessCreationParameters.h"
+#import "WKCrashReporter.h"
+#import <WebCore/RuntimeApplicationChecks.h>
+#import <wtf/text/StringHash.h>
 
 namespace WebKit {
 
+namespace SecureCoding {
+
+static bool& internalShouldCrashOnSecureCoding()
+{
+    static dispatch_once_t onceToken;
+    static bool crashOnSecureCoding { false };
+    dispatch_once(&onceToken, ^{
+        if (WebCore::isInAuxiliaryProcess())
+            return;
+
+        crashOnSecureCoding = !![[NSUserDefaults standardUserDefaults] objectForKey:@"WebKitCrashOnSecureCodingKey"];
+    });
+
+    return crashOnSecureCoding;
+}
+
+static HashSet<String>& internalClassNamesExemptFromSecureCodingCrash()
+{
+    static dispatch_once_t onceToken;
+    static NeverDestroyed<HashSet<String>> exemptClassNames;
+    dispatch_once(&onceToken, ^{
+        if (!internalShouldCrashOnSecureCoding())
+            return;
+
+        NSArray *array = [[NSUserDefaults standardUserDefaults] arrayForKey:@"WebKitCrashOnSecureCodingKey"];
```

Comment:
If the key is to be used for an array of classes to not crash on failure, I think it should have a different name.